tests/rest/admin/test_room.py

```diff
@@ -1792,6 +1795,159 @@ def test_get_joined_members_after_leave_room(self) -> None:
         self.assertEqual(403, channel.code, msg=channel.json_body)
         self.assertEqual(Codes.FORBIDDEN, channel.json_body["errcode"])
 
+    def test_timestamp_to_event(self) -> None:
```

Comment:
It probably makes sense to create an own `class` for these tests. `RoomMessageAndTimestampTestCase` is an example. Perhaps there is a better name for this.

```suggestion

class RoomMessageAndTimestampTestCase(unittest.HomeserverTestCase):
    servlets = [
        synapse.rest.admin.register_servlets,
        room.register_servlets,
        login.register_servlets,
    ]

    def prepare(self, reactor: MemoryReactor, clock: Clock, hs: HomeServer) -> None:
        self.user = self.register_user("foo", "pass")
        self.user_tok = self.login("foo", "pass")
        self.room_id = self.helper.create_room_as(user, tok=user_tok)

    def test_timestamp_to_event(self) -> None:
```

Than you do not need to recreate the user and room in every test case.